lib/workspace.rb

```diff
@@ -0,0 +1,65 @@
+require 'httparty'
+require 'dotenv'
+require_relative './user'
+require_relative './channel'
+
+# response = HTTParty.get('https://slack.com/api/channels.list?token=xoxp-my-great-key&pretty=1')
+# puts response.body, response.code, response.message, response.headers.inspect
+
+# https://slack.com/api/channels.list?token=xoxp-my-great-key&pretty=1
+class Workspace
+  include HTTParty
+  base_uri 'slack.com/api'
+
+  def initialize(api_token)
+    @options = { query: { token: api_token, pretty: 1 } }
+  end
+
+  def channels
+    response = self.class.get('/channels.list', @options).parsed_response
+    raise 'Failed to get channels' unless response['ok']
+
+    channels = response['channels'].map do |channel_map|
+      Channel.new(channel_map)
+    end
+    channels
+  end
+
+  def users
+    response = self.class.get('/users.list', @options).parsed_response
+    raise 'Failed to get users' unless response['ok']
+
```

Comment:
For this method (and `channels` above), I have two comments:

- Since there's a lot of code here that has to do with creating users, this code might fit well as a class method on `User`
- Since API calls are expensive, it might make sense to do this work once in the constructor, and save the results in instance variables

The end result might look like this:

```ruby
# user.rb
class User
  def self.list
    # move the code currently in Workspace#users to here
  end
  # ...
end

# workspace.rb
class Workspace
  attr_reader :channels, :users
  def initialize
    @channels = Channel.list
    @users = User.list
    # ...
  end
  # ...
end
```